### PR TITLE
Used in 2.4 project

### PR DESCRIPTION
**Pull request by DominicWatts**
Had to make a few adjustments to install and not affect other console commands

### CODE REVIEW

**Comment by Vinai** [comment]
Thank you for the PR! I have a few little comments, please consider them kindly.

File: Command/CustomerChangePasswordCommand.php [hunk 1]
```diff
@@ -37,26 +37,23 @@ class CustomerChangePasswordCommand extends Command
...
+     * @var AppState
+     */
+    private $state;
 
     public function __construct(
         CustomerFactory $customerFactory,
         StoreManagerInterface $storeManager,
         CustomerResource $resource,
-        AppState $appState
+        AppState $state
```
Comment (Vinai) [inline Command/CustomerChangePasswordCommand.php:50]: Why the rename?

File: Command/CustomerChangePasswordCommand.php [hunk 2]
```diff
@@ -37,26 +37,23 @@ class CustomerChangePasswordCommand extends Command
...
     ) {
         parent::__construct();
         $this->customerFactory = $customerFactory;
         $this->customerResource = $resource;
         $this->storeManager = $storeManager;
-        $this->guardAgainstSessionInitializationBug($appState);
```
Comment (Vinai) [inline Command/CustomerChangePasswordCommand.php:51]: I am absolutely with you that this should not be called in the constructor. That was a bad choice.
Moving the mutation to the `execute` method is a good idea.
However, I would like to keep the `try/catch` in place for backward compatibility as it was required by older Magento versions. I don't remember the specifics, but I do remember I added it for a reason.
So maybe keep `guardAgainstSessionInitializationBug` and just move the call to `execute?

File: composer.json [hunk 3]
```diff
@@ -2,9 +2,9 @@
   "name": "vinaikopp/module-customer-password-command",
   "description": "Set a customer password with bin/magento",
   "require": {
-    "php": "^7.0.0|^7.1.0|^7.2.0",
-    "magento/module-customer": "^100.1.1|^101.0|^102.0",
-    "magento/framework": "^100.1.1|^101.0|^102.0",
+    "php": "^7.0.0|^7.1.0|^7.2.0|^7.3.0|^7.4.0",
+    "magento/module-customer": "^100.1.1|^101.0|^102.0|^103.0",
+    "magento/framework": "^100.1.1|^101.0|^102.0^103.0",
```
Comment (Vinai) [inline composer.json]: Seems like a `||` is missing between `^102.0^103.0"`

Also, using a single pipe `|` is actually not right - it needs to be doubled: `||`.
This is obviously my mistake, but if you don't mind, could you please also fix that for the other dependencies?

**Review by Vinai — CHANGES_REQUESTED** [review]

File: Command/CustomerChangePasswordCommand.php (hunk 1 above)
Comment (DominicWatts) [reply Command/CustomerChangePasswordCommand.php:50]: Less is more

File: Command/CustomerChangePasswordCommand.php (hunk 2 above)
Comment (DominicWatts) [reply Command/CustomerChangePasswordCommand.php:51]: On execute try catch is worthless unless you use other session based logic. Perhaps you had issue in more complex scripts and just included it for safety.

File: composer.json (hunk 3 above)
Comment (DominicWatts) [reply composer.json]: Agreed. Updated.

**Comment by Vinai** [comment]
Thanks for your PR!

**Pull request merged**